src/ProductConstructionService/SubscriptionTriggerer/SubscriptionTriggererConfiguration.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Maestro.Data;
+using Maestro.DataProviders;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.DotNet.DarcLib.Helpers;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.DotNet.Kusto;
+using Azure.Storage.Queues;
+using Azure.Identity;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.DotNet.Internal.Logging;
+
+namespace SubscriptionTriggerer;
+public class SubscriptionTriggererConfiguration
+{
+    private const string ApplicationInsightsConnectionString = "APPLICATIONINSIGHTS_CONNECTION_STRING";
+    private const string AspnetcoreEnvironment = "ASPNETCORE_ENVIRONMENT";
+    private const string QueueConnectionString = "QueueConnectionString";
+    private const string ManagedIdentityClientId = "ManagedIdentityClientId";
+    private const string SqlConnectionStringUserIdPlaceholder = "USER_ID_PLACEHOLDER";
+    private const string DatabaseConnectionString = "BuildAssetRegistrySqlConnectionString";
+
+    public static IServiceCollection RegisterServices(IServiceCollection services, ITelemetryChannel telemetryChannel)
+    {
+        RegisterLogging(services, telemetryChannel);
+
+        IConfiguration config = GetConfiguration();
+        services.AddSingleton(_ => config);
+
+        string databaseConnectionString = config.GetRequiredValue(DatabaseConnectionString)
+            .Replace(SqlConnectionStringUserIdPlaceholder, config[ManagedIdentityClientId]);
+
+        services.AddBuildAssetRegistry((provider, options) =>
```

Comment:
I wonder if we should have `ProductConstructionService.Common` where we could put these registrations (SQL, Redis, Azure Queues). Maybe even the configuration extension below..

Reply:
Seems like if we change something in things like SQL, we don't want to go around all of our services and jobs

Reply:
It might make sense to do this. There will probably be a lot of registrations like this that overlap. And yeah, the Configuration Extension is something we'd probably keep adding everywhere

Reply:
Added this